packages/malloy-interfaces/thrift/malloy.thrift

```diff
@@ -429,7 +382,9 @@ union Expression {
 }
 
 struct TimeTruncationFieldReference {
-  1: required Reference field_reference, // TODO do I make this circular, more like actual grammar? e.g. TimeTruncation rather than TimeTruncationFieldReference
+  // TODO do I make this circular, referring back to `Expression`, more like actual grammar?
+  // e.g. TimeTruncation rather than TimeTruncationFieldReference
+  1: required Reference field_reference,
```

Comment:
Should `TimeTruncationFieldReference` be a `TimeTruncation` instead, having a `to_truncate: Expression` field, which is more similar to the actual grammar? Or is it okay to have this more specialized thing?

Pros:
* matches the grammar more closely
* if/when we add full (or just more) expressions to this interface, we would have to keep both `TimeTruncationFieldReference` and `TimeTruncation`

Cons: 
* makes it harder _now_, when we don't support anything other than a reference on the LHS of a time truncation, to typecheck